inginious/agent/docker_agent/__init__.py

```diff
@@ -545,14 +568,30 @@ async def handle_running_container(self, info: DockerRunningJob, future_results)
                                 hard_time_limit = min(msg["hard_time_limit"] or info.hard_time_limit, info.hard_time_limit)
                                 share_network = msg["share_network"]
                                 socket_id = msg["socket_id"]
+                                ssh = msg["ssh"]
                                 assert "/" not in socket_id  # ensure task creator do not try to break the agent :-(
                                 self._create_safe_task(self.create_student_container(info, socket_id, environment, memory_limit,
                                                                                      time_limit, hard_time_limit, share_network,
-                                                                                     write_stream))
-                            elif msg["type"] == "ssh_key":
-                                # send the data to the backend (and client)
+                                                                                     write_stream, ssh))
+                            elif msg["type"] == "ssh_debug":
+                                # send the data to the frontend (and client) to reach grading_container
                                 self._logger.info("%s %s", info.container_id, str(msg))
                                 await self.send_ssh_job_info(info.job_id, self._address_host, info.ports[22], msg["ssh_user"], msg["ssh_key"])
+
+                            elif msg["type"] == "ssh_student":
+                                # send the data to the frontend (and client) to reach student_container
+                                if info.enable_network:
+                                    info_student = None
+                                    if len(self._student_containers_running) > 0:
+                                        info_student = self._student_containers_running[msg["container_id"]]
```

Comment:
Please check that the container_id belongs to one of the student containers of this grading container.